server/main.go

```diff
@@ -100,9 +101,13 @@ func main() {
 	router.Use(LoggingMiddleware)
 	router.Use(Compress())
 
+	// repositories
+	prRepo := gormrepo.NewPullRequestRepository(database)
```

Comment:
actually i don't think the main should talk to the impl but to the interface pkg instead
like repository.NewRepository() from this it should return a repository
i don't think the main should know about the impl. of the interface itself
maybe a factory pattern or smth like this would be better 
it's nit pick IMO for small/medium size project, feel free to ignore it actually